### ISSUE TITLE
Details on eval metric aucpr implementation

### ISSUE THREAD

**Issue by Kotsakis** [dmlc/xgboost#5686]
Hello,

Since i cannot get "native" metrics (such as auc and aucpr) together with custom, i would like to implement the aucpr one. 

However i was not able to replicate it and sklearn's `metrics.auc(x=recall, y=precisionl)` (not giving me same resuluts in the agaricus set). I was wondering if you could point me to your implementation.

**Comment by trivialfis** [comment]
> Since i cannot get "native" metrics (such as auc and aucpr) together with custom

What do you mean by that?

**Comment by Kotsakis** [comment]
There are some metrics that are build in, such as logloss, auc, aucpr, error etc. 

I wish to have for model evaluation my custom metrics and aucpr. However since i cannot have the built in aucpr (in [this](https://github.com/dmlc/xgboost/issues/1125)  for example he reimplements auc) i m trying to reimplement it using sklearn . Unfortunately my results differ a lot on the agaricus data set. So i was looking at how the authors implemented to to correct my mistake.

For future i would be great if we could be able to combine custom metrics with build in ones. For example by defining the metrics as in the link and then by appending it to the watched metrics list like `eval_metric':['error', 'logloss', 'auc', 'aucpr', 'my_metric_1']`

**Comment by trivialfis** [comment]
Why not.  You can have them both.
